core/src/test/java/cucumber/runtime/ShellwordsTest.java

```diff
@@ -1,30 +1,189 @@
+            new Object[]{
+                "     --echo     --foxtrot     ",
+                asList("--echo", "--foxtrot")
+            },
+
+            // parses_single_quoted_strings
+            new Object[]{
+                "--golf 'Hotel India'",
+                asList("--golf", "Hotel India")
+            },
+
+            // parses_single_quoted_strings whitespace padding
+            new Object[]{
+                "     --juliett     'Kilo Lima'     ",
+                asList("--juliett", "Kilo Lima")
+            },
+
+            // parses_double_quoted_strings
+            new Object[]{
+                "--mike \"November Oscar\"",
+                asList("--mike", "November Oscar")
+            },
+
+            // parses_double_quoted_strings whitespace padding
+            new Object[]{
+                "     --papa      \"Quebec Romeo\"     ",
+                asList("--papa", "Quebec Romeo")
+            },
+
+            // double_quote start continue until double_quote end, and ignore single_quote early termination
+            new Object[]{
+                "--sierra \"Tango ' Uniform\"",
+                asList("--sierra", "Tango ' Uniform")
+            },
+            new Object[]{
+                "--victor \"Whiskey 'xray yankee' Zulu\"",
+                asList("--victor", "Whiskey 'xray yankee' Zulu")
+            },
+
+            // single_quote start continue until single_quote end, and ignore double_quote early termination
+            new Object[]{
+                "--alfa 'Bravo \" Charlie'",
+                asList("--alfa", "Bravo \" Charlie")
+            },
+            new Object[]{
+                "--delta 'Echo \"foxtrot golf\" hotel'",
+                asList("--delta", "Echo \"foxtrot golf\" hotel")
+            },
+
+            // double_quotes wrapped by single_quote
+            new Object[]{
+                "--india '\"Juliett Kilo\"'",
+                asList("--india", "\"Juliett Kilo\"")
+            },
+
+            // single_quote wrapped by double_quotes
+            new Object[]{
+                "--lima \"'Mike November'\"",
+                asList("--lima", "'Mike November'")
+            },
+
+            // parses_both_single_and_double_quoted_strings
+            new Object[]{
+                "--oscar \"Papa Quebec\" --romeo 'Sierra Tango'",
+                asList("--oscar", "Papa Quebec", "--romeo", "Sierra Tango")
+            },
+
+            // triple double_quote
+            // TODO is this correct with trailing empty string
+            new Object[]{
+                "--uniform \"Victor Whiskey --xray \"Yankee Zulu\"",
+                asList("--uniform", "Victor Whiskey --xray ", "Yankee", "Zulu", "\"")
+            },
+
+            // opening double_quote only with following single_quotes
+            new Object[]{
+                "--alfa \"Bravo Charlie --delta 'Echo Foxtrot'",
+                asList("--alfa", "\"", "Bravo", "Charlie", "--delta", "Echo Foxtrot")
```

Comment:
I would expect this to be `asList("--alfa", "\"Bravo", "Charlie", "--delta", "Echo Foxtrot")`